pkg/kubectl/cmd/cp/cp_test.go

```diff
@@ -587,6 +588,70 @@ func TestCopyToPod(t *testing.T) {
 	}
 }
 
+func TestCopyToPodNoPreserve(t *testing.T) {
+	tf := cmdtesting.NewTestFactory().WithNamespace("test")
+	ns := scheme.Codecs
+	codec := scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...)
+
+	tf.Client = &fake.RESTClient{
+		GroupVersion:         schema.GroupVersion{Group: "", Version: "v1"},
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			responsePod := &v1.Pod{}
+			return &http.Response{StatusCode: http.StatusNotFound, Header: cmdtesting.DefaultHeader(), Body: ioutil.NopCloser(bytes.NewReader([]byte(runtime.EncodeOrDie(codec, responsePod))))}, nil
+		}),
+	}
+
+	tf.ClientConfigVal = cmdtesting.DefaultClientConfig()
+	ioStreams, _, _, _ := genericclioptions.NewTestIOStreams()
+
+	cmd := NewCmdCp(tf, ioStreams)
+
+	srcFile, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+		t.FailNow()
+	}
+	defer os.RemoveAll(srcFile)
+
+	tests := map[string]struct {
+		expectedCmd []string
+		nopreserve  bool
+	}{
+		"copy to pod no preserve user and permissions": {
+			expectedCmd: []string{"tar", "--no-same-permissions", "--no-same-owner", "-xf", "-"},
+			nopreserve:  true,
+		},
+		"copy to pod preserve user and permissions": {
+			expectedCmd: []string{"tar", "-xf", "-"},
+			nopreserve:  false,
+		},
+	}
+
+	for name, test := range tests {
+		opts := NewCopyOptions(ioStreams)
+		src := fileSpec{
+			File: srcFile,
+		}
+		dest := fileSpec{
+			PodNamespace: "pod-ns",
+			PodName:      "pod-name",
+			File:         "foo",
+		}
+		opts.Complete(tf, cmd)
+		t.Run(name, func(t *testing.T) {
+			options := &kexec.ExecOptions{}
+			opts.NoPreserve = test.nopreserve
+			err = opts.copyToPod(src, dest, options)
+			for i, cmd := range test.expectedCmd {
```

Comment:
Use `reflect.DeepEqual` instead it read better.

Reply:
addressed thanks!